teachbooks/cli/main.py

```diff
@@ -122,108 +131,119 @@ def serve(ctx, verbose):
     If serve dir path not provided, default is `./book/_build/html`.
     Checks to see if server is already running.
     """
+    from teachbooks import BOOK_SERVE_DIR, SERVER_WORK_DIR
     from teachbooks.serve import Server
-    from teachbooks import SERVER_WORK_DIR, BOOK_SERVE_DIR
 
     if verbose > 0:
-        echo_info(f"serve command invoked.")
+        echo_info("serve command invoked.")
 
     if ctx.invoked_subcommand is None:
 
         try:
             server = Server.load(Path(SERVER_WORK_DIR))
             if verbose > 0:
-                echo_info(f" server already exists")
+                echo_info(" server already exists")
             
             stdout_summary(server)
-        except:
+        except:  # noqa: E722 TODO: handle more gracefully.
             if verbose > 0:
-                echo_info(f"no server found, creating a new one.")
-
-            dir = Path(BOOK_SERVE_DIR)
-
-            if not dir.exists():
-                echo_info(click.style("default directory not found: ", fg="yellow") + f"{dir}")
-                dir = Path(".")
-                print('            '
-                      +click.style("serving current directory: ", fg="yellow") + f"{dir}")
-                print('            '
-                      +click.style("specify a directory with: 'teachbooks serve path <path>'", fg="yellow"))
+                echo_info("no server found, creating a new one.")
+
+            serve_dir = Path(BOOK_SERVE_DIR)
+
+            if not serve_dir.exists():
+                echo_info(
+                    click.style("default directory not found: ", fg="yellow") + 
+                    f"{serve_dir}"
+                )
+
+                serve_dir = Path(".")
+                print(
+                    '            '
+                    + click.style(
+                        "serving current directory: ",
+                        fg="yellow"
+                    )
+                    + f"{serve_dir}"
+                )
+                print(
+                    '            '
+                    + click.style(
+                        "specify a directory with: 'teachbooks serve path <path>'",
+                        fg="yellow"
+                    )
+                )
                 
-            serve_path(dir, verbose)
+            serve_path(serve_dir, verbose)
 
 @serve.command()
 @click.option('-v', '--verbose', count=True)
 @click.argument("path-source",
                 type=click.Path(exists=True, file_okay=True))
 def path(path_source, verbose, no_build=False):
     """Specify relative path of directory to serve."""
-    from teachbooks.serve import Server
     from teachbooks import BUILD_DIR, SERVER_WORK_DIR
-    
-    if verbose > 0:
-        print(f"desired serve directory: {dir}")
```

Comment:
A builtin was shadowed here. So `dir` was accessed before assignment, but this didn't cause an error because `dir` is already a built-in function.

This would have printed;
```py
'desired serve directory: <built-in function dir>'
```